Makefile.envs

```diff
@@ -100,7 +100,6 @@ export MAIN_MODULE_LDFLAGS= $(LDFLAGS_BASE) \
 	-s EXPORT_NAME="'_createPyodideModule'" \
 	-s EXCEPTION_CATCHING_ALLOWED=['we only want to allow exception handling in side modules'] \
 	-s DEMANGLE_SUPPORT=1 \
-	-s USE_ZLIB=1 \
```

Comment:
I recommend that you use ``-s USE_ZLIB`` and ``-s USE_BZIP2`` here, too. It is the correct way to link with libz and libbz2.

Reply:
Personally, I prefer calling `embuilder build zlib` instead of using `-s USE_ZLIB` so I can explicitly understand what is going on behind, but yeah, for now `-s USE_ZLIB` seems more recommended...

Reply:
Python 3.11 uses the flags for compiler and linker. The ``-sUSE_FOO`` variant adds correct compiler and linker flags for various build flavors: debug/release, PIC/non-PIC, and (in the case of sqlite) single/multi-threaded. Yes, that's eight different builds for sqlite.